docs/adr/005-git-commit-structure.md

```diff
@@ -0,0 +1,162 @@
+    improvements in `src`.
+- The `banners/CHANNELNAME` structure makes it easy to navigate to the
+    banner for the current task in the file tree of the IDE/editor/CLI.
+- Dependency analysis (not implemented yet): Using tools like [Knip](https://github.com/webpro/knip)
+    to detect unused components or to analyze how many of the no-longer
+    current banners use a module or component.
+- Visualization of the banner evolution (not implemented yet): Answer question 
+    like "when did we introduce a component, how many banners use a component,
+    how many features start in mobile and go to desktop or vice versa?"
+- ??? - Question for the team: Do you see additional "use
+    cases"/scenarios where you rely on file structure or version control
+    during the campaign?
+
+### Pain points
+
+The commits to `banner/CHANNELNAME` folder work best if they consist of a
+consecutive, sequential progression of A/B tests, each test branching off
+`main` after the last test was merged. During the 2023 campaign we had
+several instances where that was not the case and banner feature branches
+branched off other banner feature branches. This led to the following
+problems:
+
+
+In a scenario where we can't branch off `main` branch for a new banner in
+the same channel, the workflow hits some snags:
+
+- We get merge and rebase conflicts when we force-push to an un-merged
+    earlier branch. It's hard to rebase the "later" branch and resolve its
+    conflicts.
+- If we merge branches in the wrong order (say Test_24, to able to work on
+    Test_25, while Test_23 is still unmerged), it becomes impossible to
+    merge earlier tests, because they would appear in the wrong order in
+    the succession of merged branches and merging would also resolve the
+    wrong conflicts.
+
+These scenarios have happened several times during the 2023 campaign.
+
+## Decision
+
+Instead of the structure `banner/CHANNELNAME`, we have
+`banner/CHANNELNAME/CAMPAIGN_NAME`. We can have several banners in
+parallel feature branches.
+
+We no longer have an "archive" commit, but in order to help make reviews easier
+we keep the 3 commit structure:
+
+1. `Prepare campaign for [CAMPAIGN_NAME]`: When creating a new test, we first
+   update the `campaign_info.toml`, and duplicate the folder from the "parent" banner
+   (i.e. a previous banner this banner is based on).
+2. `Prepare CTRL for [CAMPAIGN_NAME]`: In this commit we prepare the control banner
+   using the winner of the previous test as it's base. This will allow reviewers to
+   view a diff of the changes made to create the base from the previous test.
+3. `Implement VAR for [CAMPAIGN_NAME]`: We then implement the variant banner.
+
+We can now merge banner feature branches in any order, without conflicts
+in the banner code. The only conflict that can occur is in
+`campaign_info.toml`, but we'll resolve that conflict with the following
+rule: Always merge the version that has the higher test number.
```

Comment:
"Always merge the version that has the higher test number."
What is the meaning of _higher test number_ here?

Reply:
I'll add an explanation and example

Reply:
I've added another commit with example